src/relations/db.py

```diff
@@ -239,9 +241,23 @@ def _on_relation_joined(self, join_event: RelationJoinedEvent):
         dbs = self.charm.generate_relation_databases()
         dbs[str(join_event.relation.id)] = {"name": database, "legacy": True}
         if self.admin:
-            dbs["*"] = {"name": "*", "auth_dbname": database}
+            dbs["*"] = {"name": "*", "auth_dbname": database, "legacy": False}
         self.charm.set_relation_databases(dbs)
 
+        pgb_dbs_hash = shake_128(
+            self.charm.peers.app_databag["pgb_dbs_config"].encode()
+        ).hexdigest(16)
+        for key, data in self.charm.peers.relation.data.items():
+            # We skip the leader so we don't have to wait on the defer
+            if (
+                key != self.charm.app
+                and key != self.charm.unit
+                and data.get("pgb_dbs", "") != pgb_dbs_hash
+            ):
+                logger.debug("Not all units have synced configuration")
+                join_event.defer()
+                return
```

Comment:
Wait for all the units to be in sync (database is added). Do we want this?

Reply:
Does git blame help? I do not have the answer here.

Reply:
It's the same case as with the new relation. We don't have to do this for PGB VM, because the subordinate relations are 1:1 and each unit is adding credentials to its own databag. Only the advertised standby unit can be out of sync, so it can only be an issue if the client starts reading from the standby immediately.

Reply:
I'd prefer to keep this logic for the specific case you commented:

> Only the advertised standby unit can be out of sync, so it can only be an issue if the client starts reading from the standby immediately.

Reply:
I'd rather keep it too, better safe than sorry :)